### PR TITLE
Split task file cluster-bootstrap in two: AWS+GCP

### PR DESCRIPTION
**Pull request by lilyeyes**
Split task file cluster-bootstrap in two csp specific AWS, GCP files 

Related ticket:
[TEAM-9760](https://jira.suse.com/browse/TEAM-9760) - Split task file cluster-bootstrap in two csp specific AWS, GCP files

**VRs:**  
**hanasr AWS:**
  hanasr_aws_test_fencing_sbd_stop_kill: http://openqaworker15.qa.suse.cz/tests/303791 (passed)
  hanasr_aws_test_fencing_sbd_crash: http://openqaworker15.qa.suse.cz/tests/303799 (failed on known issue)
**hanasr GCP:**
  hanasr_gcp_test_fencing_sbd_stop_kill: http://openqaworker15.qa.suse.cz/tests/303792 (passed)
  hanasr_gcp_test_saptune_fencing_native_crash: http://openqaworker15.qa.suse.cz/tests/303800 (failed on known issue)
  
**qesap AWS**  
  qesap_aws_sapconf_test: http://openqaworker15.qa.suse.cz/tests/303793 (passed)
  qesap_aws_fencing_native_test: http://openqaworker15.qa.suse.cz/tests/303798 (passed)
**qesap GCP:**
  qesap_gcp_saptune_test: http://openqaworker15.qa.suse.cz/tests/303796 (passed)
  qesap_gcp_sbd_test: http://openqaworker15.qa.suse.cz/tests/303797 (passed)

### CODE REVIEW

**Comment by badboywj** [comment]
LGTM, thanks

**Comment by Amrysliu** [comment]
LGTM

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 1]
```diff
@@ -0,0 +1,404 @@
+---
+# This cluster-bootstrap tais an attempt of creating a standard way to build
```
Comment (lpalovsky) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:2]: s/tais/this/

**Review by lpalovsky — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml (hunk 1 above)
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:2]: Maybe it is not `this`, I don't know why use `tais`.

Comment (alvarocarvajald) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:2]: I think it's:
```
# This cluster-bootstrap playbook is an attempt of creating a standard way to build

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    stonith_action: "{{ (crm_conf_show.stdout | regex_search('stonith-action=([a-z]*)', '\\1'))[0] | default('false') }}"
+    stonith_timeout: "{{ (crm_conf_show.stdout | regex_search('stonith-timeout=([a-z0-9]*)', '\\1'))[0] | default('false') }}"
+    rsc_ip: "{{ crm_conf_show.stdout | regex_search('(rsc_ip_)') }}"
+    rsc_healthcheck_primary: "{{ crm_conf_show.stdout | regex_search('(rsc_healthcheck_primary)') }}"
+    grp_ip_hc: "{{ crm_conf_show.stdout | regex_search('(grp_ip_hc)') }}"
+    resource_stickiness: "{{ (crm_rsc_options_show.stdout | regex_search('resource-stickiness=([0-9]*)', '\\1'))[0] | default('false') }}"
+    migration_threshold: "{{ (crm_rsc_options_show.stdout | regex_search('migration-threshold=([0-9]*)', '\\1'))[0] | default('false') }}"
+    op_default_timeout: "{{ (crm_op_options_show.stdout | regex_search('timeout=([0-9]*)', '\\1'))[0] | default('false') }}"
+
+- name: Stonith command [AWS]
```
Comment (mpagot) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:217]: As far as I can see, this task `Stonith command [AWS]` is only about setting a variable `aws_stonith_cmd`  
This is something has been added https://github.com/SUSE/qe-sap-deployment/pull/257
This is fine.

But `aws_stonith_cmd`  is only used in task named `Configure AWS EC2 STONITH for sle 12sp5` that is just after a task named `Enable SBD [sbd]` that does not use it.

I propose to change order like
- Enable SBD [sbd]
- Stonith command [AWS]
- Configure AWS EC2 STONITH for sle 12sp5

Doing that will keep `Enable SBD [sbd]` in a long continuous section that AWS and GCP has in common, taht start from `Ensure cluster dependencies are installed`

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+  when:
+    - is_primary
+    - not (use_sbd | bool)
+    - ansible_distribution_major_version != '12'
+  register: stonith_config_result
+  failed_when: "'ERROR' in stonith_config_result.stderr"
+
+# This is a workaround for https://bugzilla.suse.com/show_bug.cgi?id=1231153
+# If it is fixed, this task will no longer be needed
+- name: Install python-httplib2 via zypper
```
Comment (mpagot) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:256]: This task too, it is valid for both GCP and AWS. It has been done as part of https://github.com/SUSE/qe-sap-deployment/pull/279

I propose to move this one (both in this file and GCP) just after task `Ensure cluster dependencies are installed`

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+  zypper:
+    name: python-httplib2
+    state: present
+  when: ansible_distribution_major_version == '12'
+  register: httplib2_zypper
+  ignore_errors: true
+
+- name: Set stonith-timeout [sdb]
```
Comment (mpagot) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:264]: This one is neither AWS or GCP specific. I think it has to be present in both AWS or GCP files (or in neither of them and moved to a dedicated one... but totally not sure about this last proposal)

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+  ansible.builtin.set_fact:
+    max_tasks: infinity
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=infinity")
+
+- name: Check max tasks integer
+  ansible.builtin.set_fact:
+    max_tasks_int: "{{ item | split('=') | last }}"
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=") and item is not search("DefaultTasksMax=infinity")
```
Comment (mpagot) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:152]: This line was like this from the first beginning of this (of the original) file, but eventually we can change it to (maybe in a next PR):

```suggestion
  when:
    - item is search("DefaultTasksMax=")
    - item is not search("DefaultTasksMax=infinity")
```

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+
+- name: Set DefaultTasksMax
+  ansible.builtin.lineinfile:
+    path: /etc/systemd/system.conf
+    regexp: '^DefaultTasksMax='
+    line: 'DefaultTasksMax=4096'
+    state: present
+    backup: true
+  when: max_tasks is defined and max_tasks is not match('infinity')
```
Comment (mpagot) [inline ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:161]:
```suggestion
  when:
    - max_tasks is defined
    - max_tasks is not match('infinity')
```

**Review by mpagot — APPROVED** [review]
LGTM

**Comment by lilyeyes** [comment]
@mpagot  Thanks for the comments. I will revise code in another PR since this PR is only for splitting (what I did is 99% deleting)

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+  ansible.builtin.command:
+    cmd: >-
+      crm configure property
+      $id="cib-bootstrap-options"
+      stonith-enabled=true
+  when:
+    - stonith_enabled | string | lower != 'true'
+    - is_primary
+
+- name: Disable stonith action [aws]
```
[the remark this answers is not in the document]
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:293]: Yes, I also thought to drop all aws/gcp.

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml (hunk 6 above)
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:161]: Sure, I only did deleting in this PR and did not check if coding style is good or not.

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml (hunk 5 above)
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:152]: Sure.

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml (hunk 3 above)
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:256]: Sure. I also checked pint (latest release is `sles-12-sp5-sap-v20241028-x86-64`)and the bug (latest update is `[OBSbugzilla Bot](mailto:bwiedemann+obsbugzillabot@suse.com) 2024-11-15 19:50:04 UTC`) so it seems we still need `Install python-httplib2` ATM.

File: ansible/playbooks/tasks/aws-cluster-bootstrap.yaml (hunk 2 above)
Comment (lilyeyes) [reply ansible/playbooks/tasks/aws-cluster-bootstrap.yaml:217]: Yes, really good thoughts.

**Comment by lilyeyes** [comment]
> @mpagot Thanks for the comments. I will revise code in another PR since this PR is only for splitting (what I did is 99% deleting)

Since I don't have permission to merge this PR so I updated the code according to all your comments and did new VRs. Please help to review/merge.
https://openqaworker15.qa.suse.cz/tests/303934 (hanasr_aws_test_fencing_sbd_stop_kill: passed)
https://openqaworker15.qa.suse.cz/tests/304002 (hanasr_gcp_test_fencing_sbd_stop_kill: passed)
https://openqaworker15.qa.suse.cz/tests/303936 (qesap_aws_fencing_native_test: passed)
https://openqaworker15.qa.suse.cz/tests/303937 (qesap_gcp_sbd_test: passed )

**Review by mpagot — APPROVED** [review]
LGTM

**Pull request merged**